documentation/general/design/dash-sonic-hld.md

```diff
@@ -0,0 +1,467 @@
+
+# 1 Requirements Overview
+
+## 1.1 Functional requirements
+
+At a high level the following should be supported:
+  
+- Bringup SONiC image for DEVICE_METADATA subtype - `appliance`
+- Bringup Swss/Syncd containers for switch_type - `dpu`
+- Able to program DASH objects configured via gRPC client to appliance card via SAI DASH API
+
+  Phase 1
+    - Vnet-Vnet scenario
+    - Route/LPM support
+    - Underlay IPv4 and IPv6
+    - Stateful ACL support
+    - TCP state tracking on flows
+    - Telemetry and Monitoring
+
+  Phase 2
+    - Private Link
+    - Service Tunnel
+    - Overlay IPv6 
+  
+
+## 1.2 CLI requirements
+Initial support is only for `show` commands
+
+- User should be able to show the DASH configured objects
+
+## 1.3 Warm Restart requirements
+Warm-restart support is not considered in Phase 1. TBD
+
+## 1.4 Scaling requirements
+Following are the minimal scaling requirements
+| Item                     | Expected value              |
+|--------------------------|-----------------------------|
+| VNETs                    | 1024                     |
+| ENI                      | 32 Per Card              |
+| Routes per ENI           | 100k                     |
+| NSGs per ENI             | 6                        |
+| ACLs per ENI             | 6x100K prefixes          |
+| ACLs per ENI             | 6x10K SRC/DST ports      |
+| CA-PA Mappings           | 10M                      |
+| Active Connections/ENI   | 1M                       |
+
+# 2 Packet Flows
+	
+The following section captures at a high-level on the VNET packet flow. Detailed lookup and pipeline behavior can be referenced *here*.
+
+## 2.1 Outbound packet processing pipeline
+	
+  ![dash-outbound](https://github.com/prsunny/DASH/blob/main/Assets/pkt_flow_outbound.png)
+	
+Based on the incoming packet's VNI matched against the reserved VNI assigned for VM->Appliance, the pipeline shall set the direction as TX(Outbound) and using the inner src-mac, maps to the corresponding ENI.The incoming packet will always be vxlan encapped and outer dst-ip is the appliance VIP. The pipeline shall parse the VNI, and for VM traffic, the VNI shall be a special reserved VNI. Everything else shall be treated as as network traffic(RX). Pipeline shall use VNI to differentiate the traffic to be VM (Inbound) or Network (Outbound).
+
+In the outbound flow, the appliance shall assume it is the first appliance to apply policy. It applies the outbound ACLs in three stages (VNIC, Subnet and VNET), processed in order, with the outcome being the most restrictive of the three ACLs combined. 
+
+After the ACL stage, it does LPM routing based on the inner dst-ip and applies the respective action (encap, subsequent CA-PA mapping). Finally, update the connection tracking table for both inbound and outbound. 
+	
+## 2.2 Inbound packet processing pipeline
+	
+   ![dash-intbound](https://github.com/prsunny/DASH/blob/main/Assets/pkt_flw_inbound.png)
```

Comment:
We recommend editable .svg format, e.g. using draw.io, so that the images can be maintained down the road.

Reply:
@prsunny Please, replace the current file image with this:
![dah-hld-inbound-packet-processing-pipeline](https://user-images.githubusercontent.com/2974753/160506635-979d5f98-bb4e-4cba-abfb-73d1147eb0d5.svg)